packages/server/src/utils.test.ts

```diff
@@ -20,17 +20,60 @@ describe('utils', () => {
       const activeUntil = getActiveUntilDate(dayjs(oldActiveUntil).toDate(), dayjs(anchorDate).toDate());
 
       // then
-      expect(activeUntil).toStrictEqual(dayjs(expected).toDate());
+      expect(dayjs(activeUntil).format('DD.MM.YYYY')).toStrictEqual(dayjs(expected).endOf('day').format('DD.MM.YYYY'));
     },
   );
 
-  it('should return the active period boundaries', () => {
-    const d = getPeriodFromAnchorDate(new Date('2022-01-23'), new Date('2022-12-15'));
-    expect(d.start).toStrictEqual(new Date('2022-01-15'));
-    expect(d.end).toStrictEqual(new Date('2022-02-15'));
+  // For example, a customer with a monthly subscription set to cycle on the 2nd of the month will
+  // always be billed on the 2nd.
 
-    const d1 = getPeriodFromAnchorDate(new Date('2020-02-29'), new Date('2016-12-15'));
-    expect(d1.start).toStrictEqual(new Date('2020-02-15'));
-    expect(d1.end).toStrictEqual(new Date('2020-03-15'));
-  });
+  // If a month doesn’t have the anchor day, the subscription will be billed on the last day of the month.
+  // For example, a subscription starting on January 31 bills on February 28 (or February 29 in a leap year),
+  // then March 31, April 30, and so on.
+
+  it.each([
+    // randomDate, anchorDate, start, end
+
+    // 01.01, 01.02, 01.03, 01.04
+    ['2022-01-15', '2022-01-01', '2022-01-01', '2022-01-31'],
+    ['2022-02-15', '2022-01-01', '2022-02-01', '2022-02-28'],
+
+    // 15.01, 15.02, 15.03, 15.04
+    ['2022-01-28', '2022-01-15', '2022-01-15', '2022-02-14'],
+    ['2022-02-28', '2022-01-15', '2022-02-15', '2022-03-14'],
+    ['2022-03-31', '2022-01-15', '2022-03-15', '2022-04-14'],
+    ['2022-02-05', '2022-01-15', '2022-01-15', '2022-02-14'], // randomDate 5th before anchorDate 15th
+
+    // 30.01, 28.02, 30.03, 30.04
+    ['2022-02-15', '2022-01-30', '2022-01-30', '2022-02-28'],
+    ['2022-03-15', '2022-01-30', '2022-02-28', '2022-03-30'],
+    ['2022-03-15', '2022-01-30', '2022-02-28', '2022-03-30'], // randomDate before anchorDate
+
+    // 31.01, 28.02, 31.03, 30.04
+    ['2022-02-15', '2022-01-31', '2022-01-31', '2022-02-28'], // anchorDate is 31st
+    ['2022-03-15', '2022-01-31', '2022-02-28', '2022-03-31'], // anchorDate is 31st
+    ['2022-03-15', '2022-01-31', '2022-02-28', '2022-03-31'], // randomDate before anchorDate
+
+    // 31.01, 29.02, 31.03, 30.04 (leap year)
+    ['2020-02-15', '2020-01-31', '2020-01-31', '2020-02-29'], // anchorDate is 31st
```

Comment:
@anbraten One more test case would be interesting if the anchor date is on `2020-02-29`  and I want to get the period for `2021-02-15`